python/telemetry/snet.py

```diff
@@ -168,10 +167,57 @@ def _decode(self, obj, context, path=None):
     'TargetData_ControlIsActive' / Int8ul  # flag, really
     )
 
+# Mailbox in SNET satellites
+# There are three main fields defining the Mailbox inside SNET satellites:
+#
+# Name        ByteLength  ParamType   TargetType    Interpretation
+# SysTime     4           uint32_t    float         scale: 0,001 unit: s
+# MemDataType 1           uint8_t     uint8_t       values: 0, 1, 2, 3, 4, 5
+# MemData32   32          uint8_t     Array         format: Array
+#
+# Notes:
+# * `MemDataType` refers to the slot of memory in the satellite where
+# the information is stored, which slot of memory.
+# * `MemData32` is an Array of bytes, where each byte corresponds to an ascii
+# character code. The transmission of words is done by Big-Endian order.
 
-snet = Struct(
-    'header' / SNETFrameHeader,
-    'telemetry' / Switch(lambda c: (c.header.FCIDMajor, c.header.FCIDSub), {
+
+class MemData32Adapter(Adapter):
+    def _encode(self, obj, context, path=None):
+        return bytes([x << 1 for x in bytes(
+            (obj.upper() + ' '*8)[:8], encoding='ascii')])
+
+    def _decode(self, obj, context, path=None):
+        # do we need to do right shift here?
+        return str(bytes([x for x in bytes(
+            (obj.upper() + ' '*8)[:8], encoding='ascii')]),
+                   encoding='ascii').strip()
+
+
+MemData32 = MemData32Adapter(BitsInteger(256))
+
+MailboxReceiveTelemetry = Struct(
+    'SysTime' / BitsInteger(32),
+    'MemDataType' / BitsInteger(8),
+    'MemData32' / MemData32
+    )
+
+MailboxSendTelemetry = Struct(
+    'MemDataType' / BitsInteger(8),
+    'MemData32' / BitsInteger(256)
+    )
+
+Payload = BitStruct(
```

Comment:
Why this should be a `BitStruct` rather than a `Struct`? Also, we probably don't need a single-field `Struct`. This being a field of the `snet` `Struct` as it was before would be better.

Reply:
no hard reason why using `BitStruct` instead of `Struct`. I am fine taking the change of the Payload inside the `snet` `Struct` instead.

Reply:
I don't think it is necessary to use `BitStruct` here. All the fields are `Struct`'s, so we don't need the [`Bitwise` transform](https://construct.readthedocs.io/en/latest/api/core.html#construct.core.Bitwise).